client/src/main/scala/org/scalaexercises/evaluator/implicits.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * scala-exercises-evaluator-client
+ * Copyright (C) 2015-2016 47 Degrees, LLC. <http://www.47deg.com>
+ */
+
+package org.scalaexercises.evaluator
+
+import cats.std.FutureInstances
+import cats.std.future._
+import cats.{Eval, Id, Monad, MonadError}
+import org.scalaexercises.evaluator.free.interpreters.Interpreter
+
+object implicits
+    extends Interpreter
+    with EvalInstances
+    with IdInstances
+    with FutureInstances
+
+trait EvalInstances {
+
+  implicit val evalMonadError: MonadError[Eval, Throwable] =
+    new MonadError[Eval, Throwable] {
+
+      override def pure[A](x: A): Eval[A] = Eval.now(x)
+
+      override def map[A, B](fa: Eval[A])(f: A ⇒ B): Eval[B] = fa.map(f)
+
+      override def flatMap[A, B](fa: Eval[A])(ff: A ⇒ Eval[B]): Eval[B] =
+        fa.flatMap(ff)
+
+      override def raiseError[A](e: Throwable): Eval[A] =
+        Eval.later({ throw e })
+
+      override def handleErrorWith[A](fa: Eval[A])(
+        f: Throwable ⇒ Eval[A]): Eval[A] =
+        Eval.later({
+          try {
+            fa.value
+          } catch {
+            case e: Throwable ⇒ f(e).value
+          }
+        })
+    }
+
+}
+
+trait IdInstances {
+
+  implicit def idMonadError(implicit I: Monad[Id]): MonadError[Id, Throwable] =
```

Comment:
Note that `MonadError[Id, Throwable]` is not realy a `MonadError` since `Id` values are computed eagerly and exceptions cannot be converted to values with `MonadError#attempt`. The definition of `Id` (something like `type Id[A] = A`) also makes it interact weirdly during implicit resolution.

My suggestion would be to remove this instance.